pallets/phala/src/compute/vault.rs

```diff
@@ -410,8 +410,19 @@ pub mod pallet {
 						)
 						.expect("get nft should not fail: qed.");
 						let property = &property_guard.attr;
+						if !base_pool::is_nondust_balance(property.shares) {
+							let _ = base_pool::Pallet::<T>::burn_nft(
+								&base_pool::pallet_id::<T::AccountId>(),
+								stake_pool.basepool.cid,
+								nftid,
+							);
+							return;
+						}
 						total_shares += property.shares;
 					});
+					if !base_pool::is_nondust_balance(total_shares) {
```

Comment:
`total_share` is either 0 or non-dust, because of the check you have written above.

Reply:
The total_shares is a sum of withdraw_queue shares and holding shares, the withdraw_queue shares could still be a dust, so `is_nondust_balance` is still needed.